### ISSUE TITLE
Question about non-k-uniform hyperedge

### ISSUE THREAD

**Issue by lxtlouise** [ma-compbio/Hyper-SAGNN#10]
Hi, thanks for sharing the code of the exciting work. I have several questions about the paper and the implementation. 
1. What does the argument only_rw stand for in the main file? 
2. The datasets used in the paper have three types of nodes. Did you carry out any experiments on non-k-uniform hyperedges? How do you handle the following case: The dataset is a mix of hyperedges of four nodes and hyperedges of three nodes.  The hyperedges of three nodes are the same as the ones in the given datasets. As for the hyperedges of four nodes, some of them have four different types of nodes. Some of the hyperedges of four nodes have only three data types or two data types. For such a case, what should be the input file format? The generate_H function and the generate_embeddings function are similar to the ones in DHNE. If I understand correctly, there's one incidence matrix for one type of node. Correct me if I'm wrong. 

I appreciate any help you can provide.

**Comment by ruochiz** [comment]
Hi, thanks for your interest.

1. only_rw stands for training the neural network with only random walk related loss (so doesn't include the hyperedge prediction loss, but only the hypergraph version of node2vec loss)
2. Yes. We tested the mixture of edges and hyperedges (so, k=2&3) in the Hyper-SAGNN paper, and we did another experiments in this paper (https://www.sciencedirect.com/science/article/pii/S2405471220301472). Implementation-wise, for those three-node hyperedges, you can make them "fake 4-node hyperedges" by padding them with a fake node-id, such as 0. The input format can just be the same as the rest 4 datasets, just instead of list of uniform hyperedges, you input a list of non-uniform ones.

**Comment by lxtlouise** [comment]
Thank you for the answers. They are very helpful!

Best,
Xiaoting
